File: Microservices/wowMadrid/wow-kafka-database-service/src/main/java/com/emelyAguilar/wowconsumertodatabaseservice/Event.java
package com.emelyAguilar.wowconsumertodatabaseservice;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.time.LocalTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
@Table(name= "events")
public class Event implements Serializable{
	private static final long serialVersionUID = -2723698518651206933L;
	@Id
	private String id;
	private String title;
	private String description;
	private String type;
	private Boolean free;
	private String price;
	
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss.S")
	@Column(name="date_start")
	private LocalDateTime dateStart;
	
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss.S")
	@Column(name="date_end")
	private LocalDateTime dateEnd;
	private LocalTime time;
	
	@Column(name="location_name")
	private String eventLocationName;
	
	@ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.DETACH, CascadeType.REFRESH})
    @JoinColumn(name="address")
	private Address address;
	
	@ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.DETACH, CascadeType.REFRESH})
    @JoinColumn(name="location")
	private Location location;
	
}

File: Microservices/wowMadrid/wow-dao/models/event.py
from config import db
from models.address import Address
from models.location import Location

class Event(db.Model):
    __tablename__ = 'events'

    id = db.Column(db.String(15), primary_key=True)
    title = db.Column(db.String, nullable=False)
    description = db.Column(db.String)
    type = db.Column(db.String(150))
    free = db.Column(db.Boolean)
    price = db.Column(db.String)
    date_start = db.Column(db.Date, nullable=False)
    date_end = db.Column(db.Date, nullable=False)
    time = db.Column(db.Time)
    location_name = db.Column(db.String(150))
    address = db.Column(db.Integer, db.ForeignKey('addresses.id'))
    location = db.Column(db.Integer, db.ForeignKey('locations.id'))

    def get_address_data(self):
        address_data: Address= Address.query.get(self.address)
        return address_data.dict()
    
    def get_location_data(self):
        location_data: Location= Location.query.get(self.location)
        return location_data.dict()

    def __repr__(self):
        return f"<Event(id='{self.id}', title='{self.title}', description='{self.description}', type='{self.type}', free={self.free}, price='{self.price}', date_start='{self.date_start}', date_end='{self.date_end}', time='{self.time}', location_name='{self.location_name}', address={self.address}, location={self.location})>"
    
    def dict(self):
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'type': self.type,
            'free': self.free,
            'price': self.price,
            'date_start': str(self.date_start),
            'date_end': str(self.date_end),
            'time': str(self.time),
            'location_name': self.location_name,
            'address': self.get_address_data(),
            'location': self.get_location_data()
        }